browser-test/src/admin_predicates.test.ts

```diff
@@ -1,26 +1,129 @@
-import { AdminPredicates, AdminPrograms, AdminQuestions, endSession, loginAsAdmin, logout, startSession } from './support'
+import { AdminPredicates, AdminPrograms, AdminQuestions, ApplicantQuestions, endSession, loginAsAdmin, loginAsProgramAdmin, loginAsTestUser, logout, selectApplicantLanguage, startSession, userDisplayName } from './support'
 
 describe('create and edit predicates', () => {
-  it('add a predicate', async () => {
+  it('add a hide predicate', async () => {
     const { browser, page } = await startSession();
 
     await loginAsAdmin(page);
     const adminQuestions = new AdminQuestions(page);
     const adminPrograms = new AdminPrograms(page);
 
-    // Add a program with two blocks
-    await adminQuestions.addTextQuestion('predicate-q');
-    await adminQuestions.addTextQuestion('other-q');
+    // Add a program with two screens
+    await adminQuestions.addTextQuestion('hide-predicate-q');
+    await adminQuestions.addTextQuestion('hide-other-q', 'desc', 'conditional question');
 
-    const programName = 'create predicate';
+    const programName = 'create hide predicate';
     await adminPrograms.addProgram(programName);
-    await adminPrograms.editProgramBlock(programName, 'first screen', ['predicate-q']);
-    await adminPrograms.addProgramBlock(programName, 'screen with predicate', ['other-q']);
+    await adminPrograms.editProgramBlock(programName, 'first screen', ['hide-predicate-q']);
+    await adminPrograms.addProgramBlock(programName, 'screen with predicate', ['hide-other-q']);
 
     // Edit predicate for second block
     await adminPrograms.goToEditBlockPredicatePage(programName, 'Screen 2');
     const adminPredicates = new AdminPredicates(page);
-    await adminPredicates.addPredicate('predicate-q', 'hidden if', 'text', 'is equal to', 'hide me');
-    await adminPredicates.expectVisibilityConditionEquals('Screen 2 is hidden if predicate-q\'s text is equal to "hide me"');
+    await adminPredicates.addPredicate('hide-predicate-q', 'hidden if', 'text', 'is equal to', 'hide me');
+    await adminPredicates.expectVisibilityConditionEquals('Screen 2 is hidden if hide-predicate-q\'s text is equal to "hide me"');
+
+    // Publish the program
+    await adminPrograms.publishProgram(programName);
+
+    // Switch to the applicant view and apply to the program
+    await logout(page);
+    await loginAsTestUser(page);
+    await selectApplicantLanguage(page, 'English');
+    const applicant = new ApplicantQuestions(page);
+    await applicant.applyProgram(programName);
+
+    // Initially fill out the first screen so that the next screen will be shown

```

Comment:
praise: I really appreciate all these little descriptive comments! Super useful in these longer test flows.